R/utilities-qualification-gof.R

```diff
@@ -0,0 +1,409 @@
+
+
+    plotGOFDataframe <- NULL
+    plotGOFMetadata <- list()
+    plotGOFMetadata$title <- gofPlotConfiguration$Title
+    plotGOFMetadata$sectionID <- gofPlotConfiguration$SectionId
+    plotGOFMetadata$plotTypes <- ospsuite::toPathArray(gofPlotConfiguration$PlotType)
+
+
+    plotGOFMetadata$axesSettings <- lapply(plotGOFMetadata$plotTypes, function(pltType) {
+      getAxesSettings(configurationPlan$plots$AxesSettings[[gofPlotAxesSettings[[pltType]]]])
+    })
+    names(plotGOFMetadata$axesSettings) <- plotGOFMetadata$plotTypes
+
+    plotGOFMetadata$groups <- list()
+
+
+    for (group in seq_along(gofPlotConfiguration$Groups)) {
+      gofPlotGroup <- gofPlotConfiguration$Groups[[group]]
+      caption <- gofPlotGroup$Caption
+      symbol <- gofPlotConfiguration$Groups[[group]]$Symbol
+      outputMappings <- gofPlotGroup$OutputMappings
+
+      plotGOFMetadata$groups[[caption]] <- list()
+      plotGOFMetadata$groups[[caption]]$outputMappings <- list()
+      plotGOFMetadata$groups[[caption]]$symbol <- symbol
+
+
+      for (omap in seq_along(outputMappings)) {
+        outputMapping <- outputMappings[[omap]]
+
+        projectName <- outputMapping$Project
+        simulationName <- outputMapping$Simulation
+        outputPath <- outputMapping$Output
+        color <- outputMapping$Color
+
+        observedDataPathInSimulation <- outputMapping$Output
+        observedDataSet <- outputMapping$ObservedData
+        observedDataSetFilePath <- configurationPlan$observedDataSets[configurationPlan$observedDataSets$id == outputMapping$ObservedData, ]$path
+
+        simulationFile <- configurationPlan$getSimulationPath(
+          project = projectName,
+          simulation = simulationName
+        )
+
+        simulationResultsFile <- configurationPlan$getSimulationResultsPath(
+          project = projectName,
+          simulation = simulationName
+        )
+
+        simulation <- loadSimulation(simulationFile, loadFromCache = TRUE)
+        simulationResults <- importResultsFromCSV(simulation = simulation, filePaths = simulationResultsFile)
+
+        outputs <- lapply(simulation$outputSelections$allOutputs, function(output) {
+          Output$new(output$path)
+        })
+        names(outputs) <- lapply(simulation$outputSelections$allOutputs, function(output) {
+          output$path
+        })
+        output <- outputs[[outputPath]]
+        molWeight <- simulation$molWeightFor(outputPath)
+        simulationDimension <- getQuantity(path = outputPath, container = simulation)$dimension
+
+
+
+        # Setup simulations dataframe
+        simulatedDataStandardized <- data.frame(
+          Time = simulationResults$timeValues,
+          Concentration = simulationResults$getValuesByPath(path = outputPath, individualIds = 0)
+        )
+
+        # Setup observations dataframe
+        observedDataFileData <- read.csv(file.path(inputFolder, observedDataSetFilePath), check.names = FALSE, fileEncoding = "UTF-8-BOM")
```

Comment:
Why not using the RE function `readObservedDataFile` whose default encoding is set as UTF-8 ?

Reply:
Changed